### DISCUSSION TITLE
Layout size when direction="horizontal"

### DISCUSSION THREAD

**Discussion by darylkell** [Textualize/rich#1067] [Q&A]
Hi all!

I readthedocs that when the Layout direction is horizontal, the size param is in characters width rather than in rows of height. When vertical, size is rows/lines in height.  
How then for a horizontal Layout can one determine the height of the Layout by number of lines? The only control I can see is ratio which is the scaled height, but I'd like to set a maximum height on the horizontal Layout - ie, I know the panel is only ever going to contain three lines of text so I'd like to make it 7 lines tall (2 above, 3 text, 2 below)...
I'm happy for the horizontal Layouts around this box to scale, but this one I would like to stay fixed at a certain height.

Thanks for any help in advance.  Loving rich and how extensive it has gotten so quickly!

D

**Comment by willmcgugan** [comment]
Hi,

I don't quite follow what you are asking. Perhaps draw a diagram?

**Reply by darylkell** [reply]
Thanks for the reply, Will. You're doing an awesome job on this project!

Please see attached image - I added layout.tree in one of the Panels in the hopes that it helps.  

In this instance on my dashboard, the four layouts in the bottom right (Damage, Kills, Revives, Respawns) are split horizontally inside their 'footer' layout. Because 'footer' is split horizontally, the size param for the 'damage', 'kills' etc layouts control the width of each one (an integer, representing number of characters), not the height of them.  How do I control/fix their height?

The only way I can tell is to set a size to the 'footer' layout. But in some instances I'd like to use the ratio param instead, so I can also fill the footer with other stuff and have it expand. But then I have no way to control the height of the four 'damage', 'kills' etc layouts.

Ideally it would be nice to have a height and width param for a layout which fixes it to a size (or ratio as an option, obviously) so you don't have to fix the size of its parent layout. I could then leave the parent layout ('footer' in this case) to a ratio and let it expand and contract as it needs, while the layouts inside it can be fixed if necessary
![dashboard](https://user-images.githubusercontent.com/46514397/109333463-75b49e00-789a-11eb-9121-8e1bb026ee14.PNG)
.

**Comment by willmcgugan** [comment]
Looks like that's going to be a great UI.

> not the height of them. How do I control/fix their height?

Their height is set by their parent layout, i.e. "footer". You can set a "minimum_size" with a ratio so the footer won't shrink beyond whatever you set it to.

You may be expecting the (sub) layouts to adjust their size to the content, but I'm afraid that's not what happens. What you put inside of the layout won't influence the the split. But you could adjust the layout when your content changes.
